internal/docker/client.go

```diff
@@ -35,21 +37,65 @@ const (
 )
 
 type Client struct {
-	cli    *client.Client
-	logger boshlog.Logger
-	logTag string
+	cli        *client.Client
+	logger     boshlog.Logger
+	logTag     string
+	socketPath string
+}
+
+// getDockerHost retrieves the Docker host from the current context
+func getDockerHost() (string, error) {
+	// Try to get the current context's Docker endpoint
+	cmd := exec.Command("docker", "context", "inspect", "-f", "{{.Endpoints.docker.Host}}")
```

Comment:
Use docker package for inspecting the context. Do not shell out to the docker cli.